### PR TITLE
[#4172] feat(client-java): Add client side Java REST API for Tag System

### PR DESCRIPTION
**Pull request by jerryshao**
### What changes were proposed in this pull request?

This PR proposes to add client side Java REST API for tag system.

### Why are the changes needed?

This is a part of work for tag system.

Fix: #4172 

### Does this PR introduce _any_ user-facing change?

Yes.

### How was this patch tested?

UT added.

### CODE REVIEW

**Comment by jerryshao** [comment]
@yuqi1129 @jerqi please help to review.

File: clients/client-java/src/main/java/org/apache/gravitino/client/GenericTopic.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import org.apache.gravitino.Audit;
+import org.apache.gravitino.MetadataObject;
+import org.apache.gravitino.MetadataObjects;
+import org.apache.gravitino.Namespace;
+import org.apache.gravitino.dto.messaging.TopicDTO;
+import org.apache.gravitino.messaging.Topic;
+import org.apache.gravitino.tag.SupportsTags;
+
+/** Represents a generic topic. */
+public class GenericTopic implements Topic, SupportsTagOperations {
```
Comment (yuqi1129) [inline clients/client-java/src/main/java/org/apache/gravitino/client/GenericTopic.java:32]: So what does the word `Generic` mean here? Just as it has implemented interface `SupportsTagOperations`?

Comment (jerryshao) [reply clients/client-java/src/main/java/org/apache/gravitino/client/GenericTopic.java:32]: Yeah, the purpose is to implement the `SupportsTagOperations`.

File: clients/client-java/src/main/java/org/apache/gravitino/client/SupportsTagOperations.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+interface SupportsTagOperations extends SupportsTags {
+
+  /** @return The name of the metalake. */
+  String metalakeName();
+
+  /** @return The metadata object. */
+  MetadataObject metadataObject();
+
+  /** @return The REST client. */
+  RESTClient restClient();
```
Comment (jerqi) [inline clients/client-java/src/main/java/org/apache/gravitino/client/SupportsTagOperations.java:49]: LGTM overall. Just feel concerned about making the method `restClient` public.

File: clients/client-java/src/main/java/org/apache/gravitino/client/SupportsTagOperations.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
```
Comment (jerqi) [inline clients/client-java/src/main/java/org/apache/gravitino/client/SupportsTagOperations.java:50]: If you can add an inner class to support multiple iheritance.

**Review by jerqi — APPROVED** [review]
LGTM.

File: clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoMetalake.java [hunk 4]
```diff
@@ -35,23 +36,36 @@
...
+import org.apache.gravitino.tag.TagChange;
+import org.apache.gravitino.tag.TagOperations;
 
 /**
  * Apache Gravitino Metalake is the top-level metadata repository for users. It contains a list of
  * catalogs as sub-level metadata collections. With {@link GravitinoMetalake}, users can list,
  * create, load, alter and drop a catalog with specified identifier.
  */
-public class GravitinoMetalake extends MetalakeDTO implements SupportsCatalogs {
+public class GravitinoMetalake extends MetalakeDTO implements SupportsCatalogs, TagOperations {
```
Comment (jerqi) [inline clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoMetalake.java:64]: Optional: Should we make GravitinoMetalake public?

**Comment by shaofengshi** [comment]
LGTM

**Review by shaofengshi — APPROVED** [review]

**Comment by jerryshao** [comment]
@jerqi @yuqi1129 please help to review again. Just Fix a IT issue.

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Pull request merged**